src/auth/oauth2client.ts

```diff
@@ -542,6 +554,8 @@ export class OAuth2Client extends AuthClient {
       oauth2IapPublicKeyUrl: 'https://www.gstatic.com/iap/verify/public_key',
       ...opts.endpoints,
     };
+    this.client_authentication =
+      opts.client_authentication || ClientAuthentication.ClientSecretPost;
```

Comment:
Is this supposed to be post? The RFC says the default should be basic.

Reply:
The legacy behavior is post for oauth2 endpoint.